msprime/tables.py

```diff
@@ -112,15 +96,18 @@ class NodeTable(_msprime.NodeTable):
     :vartype metadata_offset: numpy.ndarray, dtype=np.uint32
     """
 
-    def __str__(self):
+    def __str__(self, encoding='utf8', base64_metadata=True):
```

Comment:
I don't think we should change the signature of ``__str__``, as this could have some nasty unintended consequences. In this case we just have to print out **something**. Assuming utf8 and always writing it out base64 encoded is probably the right thing to do here.